src/qt/popupwin.cpp

```diff
@@ -9,12 +9,56 @@
 #include "wx/wxprec.h"
 
 #include "wx/popupwin.h"
+#include "wx/qt/private/utils.h"
+#include "wx/qt/private/winevent.h"
+
+#include <QtWidgets/QWidget>
+
+
+class wxQtPopupWindow : public wxQtEventSignalHandler<QWidget, wxPopupWindow>
+{
+public:
+    wxQtPopupWindow(wxWindow* parent, wxPopupWindow* handler);
+
+protected:
+
+    virtual void closeEvent(QCloseEvent *event) override
+    {
+        QWidget::closeEvent(event);
+    }
+};
+
+wxQtPopupWindow::wxQtPopupWindow(wxWindow* parent, wxPopupWindow* handler)
+    : wxQtEventSignalHandler<QWidget, wxPopupWindow>(parent, handler)
+{
+}
 
 wxPopupWindow::wxPopupWindow()
 {
 }
 
-wxPopupWindow::wxPopupWindow(wxWindow *WXUNUSED(parent), int WXUNUSED(flags))
+wxPopupWindow::wxPopupWindow(wxWindow *parent, int flags)
 {
+   Create(parent, flags);
 }
 
+bool wxPopupWindow::Create( wxWindow *WXUNUSED(parent), int style )
+{
+    m_qtWindow = new wxQtPopupWindow(nullptr, this);
+
+    m_qtWindow->setWindowFlag(Qt::Popup);
+    m_qtWindow->setFocusPolicy(Qt::NoFocus);
+
+    m_qtWindow->setMinimumSize(100, 100);
+
```

Comment:
Notice that we don't have to set the minimum size explicitly here. And it's here just temporarily because it appears that there is a bug in `wxQt` reporting a size of (0, 0) or (1, 1) for a `wxPanel` which prevent the popup from appearing! I'll try to debug this further...

Reply:
I think I found the solution which seems to work correctly so far!
but I need to test it more thoroughly before I can post a PR for it...

Reply:
Fixed in #23912